### ISSUE TITLE
TypeError: Cannot read property 'tapPromise' of undefined

### ISSUE THREAD

**Issue by michaelbrawn** [AnWeber/vue-cli-plugin-compression#3]
# TypeError: Cannot read property 'tapPromise' of undefined
## Desctiption

Hey @AnWeber, 

I'm having an issue with your library and what seems like a version webpack  issue. I don't have a hard dependency on webpack for my project aside from the use of this library and vue cli. I'll take a look for a previous version of your library that doesn't require webpack 5. 

I do see several errors with `yarn check` and "@vue/cli-service" when updating to webpack 5. 

I also had to install zlib which wasn't mentioned in your readme, and I'm not completely sure what to do with your "Injected webpack-chain Rules" section. 

Any tips would be appreciated. 

Thanks, 
Michael

## Examples

With Vue UI Build

```shell
-  Building for production...

 ERROR  TypeError: Cannot read property 'tapPromise' of undefined

TypeError: Cannot read property 'tapPromise' of undefined
    at /Users/username/code/my-vue-app/node_modules/vue-cli-plugin-compression/node_modules/compression-webpack-plugin/dist/index.js:349:41
    at SyncHook.eval [as call] (eval at create (/Users/username/code/my-vue-app/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:9:1)
    at SyncHook.lazyCompileHook (/Users/username/code/my-vue-app/node_modules/tapable/lib/Hook.js:154:20)
    at Compiler.newCompilation (/Users/username/code/my-vue-app/node_modules/@vue/cli-service/node_modules/webpack/lib/Compiler.js:630:30)
    at /Users/username/code/my-vue-app/node_modules/@vue/cli-service/node_modules/webpack/lib/Compiler.js:667:29
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/username/code/my-vue-app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/username/code/my-vue-app/node_modules/tapable/lib/Hook.js:154:20)
    at Compiler.compile (/Users/username/code/my-vue-app/node_modules/@vue/cli-service/node_modules/webpack/lib/Compiler.js:662:28)
    at /Users/username/code/my-vue-app/node_modules/@vue/cli-service/node_modules/webpack/lib/Compiler.js:321:11
    at Compiler.readRecords (/Users/username/code/my-vue-app/node_modules/@vue/cli-service/node_modules/webpack/lib/Compiler.js:529:11)
    at /Users/username/code/my-vue-app/node_modules/@vue/cli-service/node_modules/webpack/lib/Compiler.js:318:10
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/username/code/my-vue-app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/username/code/my-vue-app/node_modules/tapable/lib/Hook.js:154:20)
    at /Users/username/code/my-vue-app/node_modules/@vue/cli-service/node_modules/webpack/lib/Compiler.js:315:19
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/username/code/my-vue-app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/username/code/my-vue-app/node_modules/tapable/lib/Hook.js:154:20)
```

My vue.config.js
```
const CompressionPlugin = require('compression-webpack-plugin');
const zlib = require('zlib');

module.exports = {
  publicPath: '',
  chainWebpack(config) {
    config.plugins.delete('prefetch');
    config.plugin('CompressionPlugin').use(CompressionPlugin);
  },
  pluginOptions: {
    compression: {
      brotli: {
        filename: '[file].br[query]',
        algorithm: 'brotliCompress',
        include: /\.(js|css|html|svg|json)(\?.*)?$/i,
        compressionOptions: {
          params: {
            [zlib.constants.BROTLI_PARAM_QUALITY]: 11,
          },
        },
        minRatio: 0.8,
      },
      gzip: {
        filename: '[file].gz[query]',
        algorithm: 'gzip',
        include: /\.(js|css|html|svg|json)(\?.*)?$/i,
        minRatio: 0.8,
      },
    },
  },
};
```

yarn check  with webpack 5. 
```
error "@vue/cli-plugin-eslint#eslint@>= 1.6.0 < 7.0.0" doesn't satisfy found match of "eslint@7.30.0"
warning "@vue/cli-service#html-webpack-plugin@^3.2.0" could be deduped from "3.2.0" to "html-webpack-plugin@3.2.0"
error "vue-cli-plugin-compression#webpack@^4.0.0" doesn't satisfy found match of "webpack@5.44.0"
warning "@vue/cli-service#@soda/friendly-errors-webpack-plugin#webpack@^4.0.0 || ^5.0.0" could be deduped from "5.44.0" to "webpack@5.44.0"
warning "@vue/cli-service#@vue/preload-webpack-plugin#html-webpack-plugin@>=2.26.0" could be deduped from "3.2.0" to "html-webpack-plugin@3.2.0"
warning "@vue/cli-service#@vue/preload-webpack-plugin#webpack@>=4.0.0" could be deduped from "5.44.0" to "webpack@5.44.0"
warning "@vue/cli-plugin-babel#babel-loader#webpack@>=2" could be deduped from "5.44.0" to "webpack@5.44.0"
warning "@vue/cli-service#copy-webpack-plugin#webpack@^4.0.0 || ^5.0.0" could be deduped from "5.44.0" to "webpack@5.44.0"
warning "@vue/cli-service#css-loader#webpack@^4.0.0 || ^5.0.0" could be deduped from "5.44.0" to "webpack@5.44.0"
error "@vue/cli-plugin-eslint#eslint-loader#eslint@>=1.6.0 <7.0.0" doesn't satisfy found match of "eslint@7.30.0"
error "vue-cli-plugin-vuetify#null-loader#webpack@^4.3.0" doesn't satisfy found match of "webpack@5.44.0"
warning "@vue/cli-service#vue-loader#webpack@^3.0.0 || ^4.1.0 || ^5.0.0-0" could be deduped from "5.44.0" to "webpack@5.44.0"
warning "@vue/cli-service#webpack-dev-server#webpack@^4.0.0 || ^5.0.0" could be deduped from "5.44.0" to "webpack@5.44.0"
warning "@vue/cli-service#webpack-dev-server#webpack-dev-middleware#webpack@^4.0.0 || ^5.0.0" could be deduped from "5.44.0" to "webpack@5.44.0"
info Found 10 warnings.
error Found 4 errors.
```

yarn check with webpack 4
```
error "@vue/cli-plugin-eslint#eslint@>= 1.6.0 < 7.0.0" doesn't satisfy found match of "eslint@7.30.0"
error "compression-webpack-plugin#webpack@^5.1.0" doesn't satisfy found match of "webpack@4.46.0"
error "@vue/cli-plugin-eslint#eslint-loader#eslint@>=1.6.0 <7.0.0" doesn't satisfy found match of "eslint@7.30.0"
error Found 3 errors.
info Visit https://yarnpkg.com/en/docs/cli/check for documentation about this command.
```

**Comment by AnWeber** [comment]
When was there an update for vue cli that supports webpack 5? In this case I need to deploy a new version that uses the latest webpack bundle analyzer (>7)

**Comment by michaelbrawn** [comment]
Ah, I do see they're still only in [beta](https://github.com/vuejs/vue-cli/releases/tag/v5.0.0-beta.2). 

I've downgraded to `"compression-webpack-plugin": "6.1.1"`  which seems to be working.

**Comment by michaelbrawn** [comment]
Also updating to the Vue CLI beta resolves the issue. I can mark this as closed if that's easiest.

**Comment by AnWeber** [comment]
Yes, I meant `compression-webpack-plugin`:-) I have a second vue-cli plugin and there is also the problem. I will create a prerelease for webpack 5, then you can use this also already with it

**Comment by AnWeber** [comment]
I have released a version 2.0.0-rc.0 which will be compatible with webpack 5.
